### DISCUSSION TITLE
bulkTagAssets failing with {"message":"Failed to bulk tag assets","error":"Internal Server Error"

### DISCUSSION THREAD

**Discussion by dlang123** [immich-app/immich#14421] [Q&A]
I have written a python script that searches for certain album names and applys a tag to that album name basically.  It is a little more involved than that.  When I look at the URL and payload data everything has the correct uuid. I recieve the following error however:
 {"message":"Failed to bulk tag assets","error":"Internal Server Error","statusCode":500,

Why do I receive this error?

Here is the basic function call that does it

```
def add_tag_to_assets(album_id, tag_name,aj_tag_id):
    headers_data = {
        "Content-Type": "application/json",
        "Accept": "application/json",
        "x-api-key": API_TOKEN
    }
    url = f"{IMMICH_URL}/api/tags/assets"
    data = {
        "assetIds": [],
        "tagIds": [aj_tag_id]  # Replace with the ID of the "AJ" tag
        
    }

    # Get asset IDs from the album
    album_url = f"{IMMICH_URL}/api/albums/{album_id}"
    album_response = requests.get(album_url, headers=headers)
    album_assets = album_response.json()

    for asset in album_assets['assets']:
        # print(asset)
        data["assetIds"].append(asset["id"])

    print(data)
    response = requests.put(url, headers=headers_data, json=data)
    if response.status_code != 200:
        print(f"Error adding tag to album {album_id}: {response.text}")
    else:
        print(f"Added {aj_tag_id} tag to album {album_id}: {response.text}")
```

**Comment by bo0tzz** [comment]
There'll be an actual error message in the server logs.

**Comment by dlang123** [comment]
ERROR [Api:ErrorInterceptor~4l25r1pj] Database error: QueryFailedError:
duplicate key value violates unique constraint
"PK_ef5346fe522b5fb3bc96454747e"

so is this saying that basically that tag is already applied?

**Reply by dlang123** [reply]
I actually modified my use case and quit using bulktag and used just the tagAssets so I can see in the return that there are duplicates for some of the tags.  For my use case just changing to the tagAssets will work for now so will just let it run.
